src/components/views/DownloadModModal.vue

```diff
@@ -15,56 +15,14 @@
-            <template v-slot:header>
-                <h2 class='modal-title' v-if="thunderstoreMod !== null">
-                    Select a version of {{thunderstoreMod.getName()}} to download
-                </h2>
-            </template>
-            <template v-slot:body>
-                <p>It's recommended to select the latest version of all mods.</p>
-                <p>Using outdated versions may cause problems.</p>
-                <br/>
-                <div class="columns is-vcentered">
-                    <template v-if="currentVersion !== null">
-                        <div class="column is-narrow">
-                            <select class="select" disabled="true">
-                                <option selected>
-                                    {{currentVersion}}
-                                </option>
-                            </select>
-                        </div>
-                        <div class="column is-narrow">
-                            <span class="margin-right margin-right--half-width"><span class="margin-right margin-right--half-width"/> <i class='fas fa-long-arrow-alt-right'></i></span>
-                        </div>
-                    </template>
-                    <div class="column is-narrow">
-                        <select class='select' v-model='selectedVersion'>
-                            <option v-for='(value, index) in versionNumbers' :key='index' v-bind:value='value'>
-                                {{value}}
-                            </option>
-                        </select>
-                    </div>
-                    <div class="column is-narrow">
-                        <span class="tag is-dark" v-if='selectedVersion === null'>
-                            You need to select a version
-                        </span>
-                        <span class="tag is-success" v-else-if='recommendedVersion === selectedVersion'>
-                            {{selectedVersion}} is the recommended version
-                        </span>
-                        <span class="tag is-success" v-else-if='versionNumbers[0] === selectedVersion'>
-                            {{selectedVersion}} is the latest version
-                        </span>
-                        <span class="tag is-danger" v-else-if='versionNumbers[0] !== selectedVersion'>
-                            {{selectedVersion}} is an outdated version
-                        </span>
-                    </div>
-                </div>
-            </template>
-            <template v-slot:footer>
-                <button class="button is-info" @click="downloadThunderstoreMod()">Download with dependencies</button>
-            </template>
-        </ModalCard>
+        <DownloadModVersionSelectModal
+            :versionNumbers="versionNumbers"
+            :selected-version="selectedVersion"
+            :currentVersion="currentVersion"
+            :recommendedVersion="recommendedVersion"
```

Comment:
This and the next comment are about the same thing but I write them separately to make it clearer what my train of thought was. Read them both before doing changes.

It looks to me that these four props are passed to child component somewhat unnecessarily. If you look at what `getModVersions` does, it seems that it should be a part of the child component, not the parent. If the watched method is moved to the child, only thing the parent cares about is the selected version.

Also the way `selectedVersion` and `setSelectedVersion` is passed here is a bit React-ish. It will probably work without problems, but using emits would be more in line with Vue. You could take a look at how `CategoryFilterModal` and `CategorySelectorModal` currently interact. It's not an exact match so don't copy it as-is, and read the Vue docs about events/emits to get the whole picture.

Reply:
Should be better now (in the next commit).
I banged my head against the wall trying to do this exact thing last night for hours, but it just wouldn't work. Well, at least it works now.